int32 channel
int32 prev
---
int32 value
